src/NCronJob/Execution/JobExecutor.cs

```diff
@@ -73,9 +73,9 @@ public async Task RunJob(JobRun run, CancellationToken stoppingToken)
         catch (Exception exc) when (exc is not OperationCanceledException or AggregateException)
         {
             LogExceptionHandlerError(runContext.JobType);
-            runContext.JobRun.NotifyStateChange(JobStateType.Faulted, exc);
             await NotifyExceptionHandlers(runContext, exc, stoppingToken);
             await AfterJobCompletionTask(runContext, exc, linkedCts.Token);
+            runContext.JobRun.NotifyStateChange(JobStateType.Faulted, exc);
```

Comment:
On success, dependent jobs are created before the root job is marked as Completed.
We also need that behavior when a fault happens.
Otherwise, some concurrency issue may happen and the root job being completed, this may also trigger the completion of the orchestration. Which we do not want ;-)

Reply:
Oh interesting bit - and yes. The parent job should be "done" (as in: Its observable state is either finished/completed or faulted)